File: HW_2/transport.cpp
#include "transport.h"

transport::transport() {

}

int RandInt(int min, int max);

// Calculate time to cover the distance.
double transport::MeasureTime() {
    if (this->velocity == 0) {
        return 0;
    }
    return this->distance / this->velocity;
}

// Read transport information from file.
void transport::Read(FILE *file) {
    fscanf(file, "%d", &this->velocity);
    fscanf(file, "%lf", &this->distance);
}

//  Write transport information to file.
void transport::Write(FILE *file) {
    fprintf(file, "%d%c%f", this->velocity, ' ', this->distance);
}

// Generate random transport.
void transport::Generate() {
    this->velocity = RandInt(1, 500);
    this->distance = RandInt(1, 1000);
}

File: HW_3/Transport.py
import random


class Transport:
    def __init__(self, params=None):
        if params is None:
            self.distance = 0
            self.velocity = 0
        else:
            self.velocity = int(params[0])
            self.distance = float(params[1])

    def measure_time(self) -> float:
        if not self.velocity:
            return 1e100 + self.distance
        return self.distance / self.velocity

    def generate(self):
        self.distance = random.randint(1, 1000)
        self.velocity = random.randint(1, 400)
        return self

    def __str__(self):
        return f"{round(self.measure_time(), 3)} {self.velocity} {self.distance}"
